### PR TITLE
Reimplement the `cargo func run` command. 

### PR DESCRIPTION
**Pull request by peterhuene**
This PR reimplements the `cargo func run` command to use `cargo run` to
generate a script root and then `func host start` from the Azure Functions Core
Tools to run the application natively.

This removes the requirement for using Docker to build and run the application
locally.

Additionally, the `cargo func build` command has been removed in favor of `cargo build`.

### CODE REVIEW

File: azure-functions/src/lib.rs [hunk 1]
```diff
@@ -187,118 +180,265 @@ fn initialize_app(
...
+        println!("Creating function directory '{}'.", function_dir.display());
+    }
 
-        let destination_file = function_dir.join(
-            source_file
-                .file_name()
-                .expect("expected the source file to have a file name"),
-        );
+    fs::create_dir(&function_dir).unwrap_or_else(|_| {
+        panic!(
```
Comment (peterhuene) [inline azure-functions/src/lib.rs]: Should remove this panic in favor of changing this to return `Result` like the other functions.

File: azure-functions-sdk/src/commands/run.rs [hunk 2]
```diff
@@ -45,10 +44,16 @@ impl<'a> Run<'a> {
...
-                    .value_name("IMAGE")
-                    .help("The image of the Azure Function application to run. Default is based off the crate name.")
-                    .index(1),
+                Arg::with_name("script_root")
+                    .long("script-root")
+                    .short("r")
+                    .value_name("ROOT")
+                    .help("The directory to use for the Azure Functions application script root. Default is a temporary directory."),
+            )
+            .arg(Arg::with_name("cargo_option")
```
Comment (rylev) [inline azure-functions-sdk/src/commands/run.rs]: nit: maybe should be called `cargo_options`?

Comment (peterhuene) [reply azure-functions-sdk/src/commands/run.rs]: I'll fix.

**Comment by peterhuene** [comment]
Weird, with the recent changes it shouldn't even be using go to build 🤔.  I wonder if it's a cached CMake thing since it previously configured with go to build BoringSSL.  Since switching to OpenSSL, the golang dependency was removed.

**Comment by rylev** [comment]
I didn't realize it was no longer using BoringSSL. That is truly weird. I'll take a look again to see what could possibly be the issue.

**Comment by peterhuene** [comment]
@rylev I've pushed up the requested change and it dismissed your review (I've since turned that off; we don't use that for the .NET Core repos either).  Would you mind taking a look at the new commit and signing off again?  Thanks!

**Comment by peterhuene** [comment]
Ping @rylev 😁

**Comment by rylev** [comment]
@peterhuene trying to get it to run now. Just wanted to make sure the weird thing I saw with grpc wasn't an actual issue.

**Comment by peterhuene** [comment]
Cool! The Windows fixes PR has all the changes if you want to checkout that one to test in one go.

**Pull request merged**